### PR TITLE
Update CodeAnalysis config to include recently-added rules

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/tools/illink/src/linker/Mono.Linker.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <ContractProject>$(MSBuildThisFileDirectory)ref\Mono.Linker.csproj</ContractProject>
     <RollForward>Major</RollForward>
     <UseAppHost>false</UseAppHost>
-    <NoWarn>$(NoWarn);NU5131</NoWarn>
+    <NoWarn>$(NoWarn);NU5131;CA1866</NoWarn>
```
Comment (buyaa-n) [inline src/tools/illink/src/linker/Mono.Linker.csproj:22]: FYI there is more diagnostics found on other mono projects

File: eng/CodeAnalysis.src.globalconfig [hunk 2]
```diff
@@ -489,6 +489,15 @@ dotnet_diagnostic.CA1863.severity = suggestion
...
 dotnet_diagnostic.CA1864.severity = warning
 
+# CA1865: Use char overload
+dotnet_diagnostic.CA1865.severity = warning
+
+# CA1866: Use char overload
+dotnet_diagnostic.CA1866.severity = warning
+
+# CA1867: Use char overload
+dotnet_diagnostic.CA1867.severity = warning
```
Comment (buyaa-n) [inline eng/CodeAnalysis.src.globalconfig:499]: Did not know they were missing :)

File: src/coreclr/tools/aot/ILCompiler.Compiler/ILCompiler.Compiler.csproj [hunk 3]
```diff
@@ -4,6 +4,7 @@
     <AssemblyName>ILCompiler.Compiler</AssemblyName>
     <TargetFramework>$(NetCoreAppToolCurrent)</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <NoWarn>$(NoWarn);CA1866;CA1867</NoWarn>
```
Comment (buyaa-n) [inline src/coreclr/tools/aot/ILCompiler.Compiler/ILCompiler.Compiler.csproj:7]: Just curious, why these disabled in some projects

**Review by buyaa-n — APPROVED** [review]
Thank you!

**Comment by kg** [comment]
This made WASM startup faster, thanks @stephentoub :)

**Comment by stephentoub** [comment]
> This made WASM startup faster, thanks @stephentoub :)

Interesting. That's a nice side effect, though it wasn't the primary intent :) Do you know why it got faster? Only explanation I could think of were some of the string-to-char changes could have resulted possibly in globalization code not being needed until later?

**Comment by kg** [comment]
There was a search for `"."` instead of `'.'` that was fully loading in ICU during startup.

**Pull request merged**